libsycl/include/sycl/__impl/info/platform.def

```diff
@@ -0,0 +1,8 @@
+#ifndef __SYCL_PARAM_TRAITS_SPEC
+static_assert(false, "__SYCL_PARAM_TRAITS_SPEC is required but not defined");
+#endif
+
+// 4.6.2.4. Information descriptors
+__SYCL_PARAM_TRAITS_SPEC(platform, version, std::string, OL_PLATFORM_INFO_VERSION)
+__SYCL_PARAM_TRAITS_SPEC(platform, name, std::string, OL_PLATFORM_INFO_NAME)
+__SYCL_PARAM_TRAITS_SPEC(platform, vendor, std::string, OL_PLATFORM_INFO_VENDOR_NAME)
```

Comment:
Subjective, but I'm not a fan of this approach for two reasons:

1) Smart use of preprocessor in distributable header files (i.e., `libsycl/include` instead of `libsycl/source/**`).
2) `OL_*` are implementation details and don't need to be present in those distributable headers.

I think avoiding it doesn't result in too much code duplication: https://github.com/llvm/llvm-project/commit/823c76576b4992a3aeab9ce289bd13e73a185b4b, but I understand that this is subjective.

Reply:
> Subjective, but I'm not a fan of this approach for two reasons:
> 
> 1. Smart use of preprocessor in distributable header files (i.e., `libsycl/include` instead of `libsycl/source/**`).
> 2. `OL_*` are implementation details and don't need to be present in those distributable headers.

I don't understand the problem with using pre-processor in distributed header files, but I agree that mapping from SYCL API to liboffload API should be done in the library source code.